packages/core-utils/src/list-utils.test.ts

```diff
@@ -0,0 +1,19 @@
+import { append } from './list-utils';
+
+describe('append', () => {
+  it('adds item to list of items', () => {
+    // Act
+    const result = append(3)([1, 2]);
+
+    // Assert
+    expect(result).toEqual([1, 2, 3]);
+  });
+
+  it('adds item to an empty list', () => {
+    // Act
+    const result = append('a')([]);
+
+    // Assert
+    expect(result).toEqual(['a']);
+  });
```

Comment:
an additional test to assert that immutable pattern is held up would be nice... simple equality (reference value) of result and orignalValue should be false.